### PR TITLE
test_cephfs_shell: refactor test_put_and_get_without_target_directory

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
The refactor intends to simplify the test and make it more readable,

It also changes how this test creates directory -- tempfile.mkdtemp is
used instead of coreutils mkdir to avoid bug reported here -
https://tracker.ceph.com/issues/43567.

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
@batrick Do we need a tracker ticket for this?

**Comment by rishabh-d-dave** [comment]
Depends on https://github.com/ceph/ceph/pull/32612

**Comment by rishabh-d-dave** [comment]
Got a valid failure - http://pulpito.ceph.com/rishabh-2020-02-13_19:08:14-fs-wip-rishabh-fs-refactor-test_put_and_get-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
Tests passed - http://pulpito.ceph.com/rishabh-2020-02-14_05:20:53-fs-wip-rishabh-fs-refactor-test_put_and_get-distro-basic-smithi/

**Comment by gregsfortytwo** [comment]
Dropping needs-review label until the depends-on #32612 is ready to go.

**Comment by rishabh-d-dave** [comment]
Ran a quick test, it succeeded - http://pulpito.ceph.com/rishabh-2020-04-17_20:32:30-fs-wip-fs-refactor-test_put_and_get-distro-basic-smithi/

File: qa/tasks/cephfs/test_cephfs_shell.py [hunk 1]
```diff
@@ -257,60 +260,37 @@ def test_rmdir_p_dir_with_file(self):
...
-        o = self.get_cephfs_shell_cmd_output('!stat p1 || echo $?')
-        log.info("cephfs-shell output:\n{}".format(o))
-        self.validate_stat_output(o)
-
-        o = self.get_cephfs_shell_cmd_output('!stat p1/dump1 || echo $?')
-        log.info("cephfs-shell output:\n{}".format(o))
-        self.validate_stat_output(o)
+    def test_without_target_dir(self):
+        """
+        Test that both, put and get commands, without target path.
```
Comment (varshar16) [inline qa/tasks/cephfs/test_cephfs_shell.py:265]:
```suggestion
        Test put and get commands without target path.
```

File: qa/tasks/cephfs/test_cephfs_shell.py [hunk 2]
```diff
@@ -257,60 +260,37 @@ def test_rmdir_p_dir_with_file(self):
...
+        """
+        tempdir = self.mount_a.client_remote.mkdtemp()
+        tempdirname = path.basename(tempdir)
+        dump1, dump2, dump3 = ('dump1', 'dump2', 'dump3')
```
Comment (varshar16) [inline qa/tasks/cephfs/test_cephfs_shell.py:269]: Save directly as set instead.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/test_cephfs_shell.py:269]: Let's skip this? I only have a list of strings and not a list of key-value pairs.

Comment (varshar16) [reply qa/tasks/cephfs/test_cephfs_shell.py:269]: True but you are using the same set of strings many times. Instead we can directly use a set or list of thoe string values.
`dump_file = ('dump1', 'dump2', 'dump3')`
Later append tempdirname to dump_file.

File: qa/tasks/cephfs/test_cephfs_shell.py [hunk 3]
```diff
@@ -257,60 +260,39 @@ def test_rmdir_p_dir_with_file(self):
...
-
-        self.run_cephfs_shell_cmd('!rm -rf p1')
-        o = self.get_cephfs_shell_cmd_output("get p1")
-        o = self.get_cephfs_shell_cmd_output('!stat p1 || echo $?')
-        log.info("cephfs-shell output:\n{}".format(o))
-        self.validate_stat_output(o)
+        for file_ in files[ : len(files) - 1]:
+            size = files.index(file_) + 1
+            ofarg = 'of=' + path.join(tempdir, file_)
+            bsarg = 'bs=' + str(size) + 'M'
```
Comment (varshar16) [inline qa/tasks/cephfs/test_cephfs_shell.py:274]:
```suggestion
        for idx, file_ in enumerate(files[: -1]):
            ofarg = 'of=' + path.join(tempdir, file_)
            bsarg = 'bs=' + str(idx+1) + 'M'
```

File: qa/tasks/cephfs/test_cephfs_shell.py [hunk 4]
```diff
@@ -257,60 +260,39 @@ def test_rmdir_p_dir_with_file(self):
...
-        log.info("mount_a output:\n{}".format(o))
-        o = self.mount_a.stat('p1/dump3')
-        log.info("mount_a output:\n{}".format(o))
-
-        self.run_cephfs_shell_cmd('!rm -rf p1')
-        o = self.get_cephfs_shell_cmd_output("get p1")
-        o = self.get_cephfs_shell_cmd_output('!stat p1 || echo $?')
-        log.info("cephfs-shell output:\n{}".format(o))
-        self.validate_stat_output(o)
+        for i, file_ in enumerate(files[ : len(files) - 1]):
```
Comment (varshar16) [inline qa/tasks/cephfs/test_cephfs_shell.py:271]: This can simply be:
```suggestion
        for i, file_ in enumerate(files[ : - 1]):
```

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/test_cephfs_shell.py]: Yeah, I keep forgetting about negative indices. Fixed; thanks!

**Comment by rishabh-d-dave** [comment]
@varshar16 @gregsfortytwo Testing with teuth was successful - http://pulpito.ceph.com/rishabh-2020-04-27_10:27:30-fs-fs-rishabh-refactor-distro-basic-smithi

**Comment by gregsfortytwo** [comment]
Remember to update the tags and PR assignment when it's ready to get picked up in integration. :)

**Comment by varshar16** [comment]
jenkins retest this please

**Comment by varshar16** [comment]
Running the make check again as the flake8 ci testing is fixed now.

**Comment by rishabh-d-dave** [comment]
@batrick Can we please merge this PR?

**Comment by rishabh-d-dave** [comment]
@gregsfortytwo Any updates on the status for the testing of this PR?

**Pull request merged**